priv/migration_tasks/prod/1.5.14@add_geopatch_node_transactiosn.ex

```diff
@@ -0,0 +1,159 @@
+
+        case send_node_transaction() do
+          :ok ->
+            Logger.info("Migration complete for: #{Base.encode16(node_pk)}")
+            {:halt, :ok}
+
+          {:error, reason} ->
+            Logger.error("Migration failed for: #{Base.encode16(node_pk)}")
+            {:halt, {:error, reason}}
+        end
+      else
+        PubSub.register_to_new_transaction_by_type(:node)
+
+        receive do
+          {:new_transaction, address, :node, _timestamp} ->
+            with {:ok, %Transaction{previous_public_key: previous_pk} = transaction} <-
+                   TransactionChain.get_transaction(address) do
+              first_pk = TransactionChain.get_first_public_key(previous_pk)
+
+              if first_pk == node_pk do
+                process_transaction(transaction)
+              else
+                {:cont, :ok}
+              end
+            else
+              {:error, reason} ->
+                Logger.error(
+                  "Failed to fetch transaction for address #{Base.encode16(address)}: #{inspect(reason)}"
+                )
+
+                {:cont, :ok}
+            end
+        after
+          60_000 ->
+            Logger.error("Timeout waiting for updates from: #{Base.encode16(node_pk)}")
+
+            PubSub.unregister_to_new_transaction_by_type(:node)
+            {:cont, :ok}
+        end
+      end
+    end)
+  end
+
+  defp process_transaction(transaction) do
+    case check_geopatch_status(transaction) do
+      :geopatch_set ->
+        PubSub.unregister_to_new_transaction_by_type(:node)
+        {:cont, :ok}
+
+      :waiting_for_update ->
+        PubSub.unregister_to_new_transaction_by_type(:node)
+        {:cont, :ok}
+    end
+  end
+
+  defp check_geopatch_status(%Transaction{data: %TransactionData{content: content}}) do
+    with {:ok, _ip, _p2p_port, _http_port, _transport, _last_reward_address, _origin_public_key,
+          _key_certificate, _mining_public_key,
+          geo_patch} <- Node.decode_transaction_content(content) do
+      if geo_patch do
+        :geopatch_set
+      else
+        :waiting_for_update
+      end
+    else
+      _ ->
+        Logger.error("Could not find a set geopatch in the last transaction for this node.")
+
+        :waiting_for_update
+    end
+  end
+
+  defp send_node_transaction() do
+    %Node{
+      ip: ip,
+      port: port,
+      http_port: http_port,
+      transport: transport,
+      reward_address: reward_address,
+      origin_public_key: origin_public_key
+    } = P2P.get_node_info()
+
+    geopatch = Archethic.P2P.GeoPatch.from_ip(ip)
+
+    mining_public_key = Crypto.mining_node_public_key()
+    key_certificate = Crypto.get_key_certificate(origin_public_key)
+
+    genesis_address = Crypto.first_node_public_key() |> Crypto.derive_address()
+
+    {:ok, %Transaction{data: %TransactionData{code: code}}} =
+      TransactionChain.get_last_transaction(genesis_address, data: [:code])
```

Comment:
`P2P.get_node_info` returns the last_address of the node in the Node struct. You could directly use this last_address using the function `TransactionChain.get_transaction()`